### ISSUE TITLE
Difference between the GRPO implementation in this work and the original GRPO

### ISSUE THREAD

**Issue by JuzhengMiao** [Liuziyu77/Visual-RFT#250]
I found that the implementation of GRPO in grpo_trainer.py of this work seems to have some differences compared to the original GRPO algorithm. For example, it seems that the clip operation is not taken, and the old logits of the tokens is not used.

**Comment by childeyemax** [comment]
per_token_loss = torch.exp(per_token_logps - per_token_logps.detach()) * advantages.unsqueeze(1)
https://github.com/Liuziyu77/Visual-RFT/blob/main/src/virft/src/open_r1/trainer/grpo_trainer.py#L465

$\pi_\theta/\pi_{ref}$ seems to be approximated as 1.

**Comment by JuzhengMiao** [comment]
> per_token_loss = torch.exp(per_token_logps - per_token_logps.detach()) * advantages.unsqueeze(1) https://github.com/Liuziyu77/Visual-RFT/blob/main/src/virft/src/open_r1/trainer/grpo_trainer.py#L465
> 
> π θ / π r e f seems to be approximated as 1.

Yes. I guess this is the case when the number of iterations of PPO equals to 1.
